Sources/SwiftProtobuf/JSONEncoder.swift

```diff
@@ -53,68 +129,116 @@ public struct JSONEncoder {
+            if let v = Int64(exactly: value) {
+                appendInt(value: v)
             } else {
+                let s = String(value)
                 append(text: s)
             }
         }
     }
-    mutating func putInt64(value: Int64, quote: Bool) {
-        // Always quote integers with abs value > 2^53
-        if quote || value > 0x1FFFFFFFFFFFFF || value < -0x1FFFFFFFFFFFFF {
-            append(text: "\"" + String(value) + "\"")
+    private mutating func appendUInt(value: UInt64) {
+        if value >= 10 {
+            appendUInt(value: value / 10)
+        }
+        data.append(asciiZero + UInt8(value % 10))
+    }
+    private mutating func appendInt(value: Int64) {
+        if value < 0 {
+            data.append(asciiMinus)
+            // This is the twos-complement negation of value,
+            // computed in a way that won't overflow a 64-bit
+            // signed integer.
+            appendUInt(value: 1 + ~UInt64(bitPattern: value))
         } else {
-            append(text: String(value))
+            appendUInt(value: UInt64(bitPattern: value))
         }
     }
-    mutating func putUInt64(value: UInt64, quote: Bool) {
-        if quote || value > 0x1FFFFFFFFFFFFF { // 2^53 - 1
-            append(text: "\"" + String(value) + "\"")
+    internal mutating func putEnumInt(value: Int) {
+        appendInt(value: Int64(value))
+    }
+    internal mutating func putInt64(value: Int64) {
+        data.append(asciiDoubleQuote)
+        appendInt(value: value)
+        data.append(asciiDoubleQuote)
+    }
+    internal mutating func putInt32(value: Int32) {
+        if isMapKey {
+            data.append(asciiDoubleQuote)
+            appendInt(value: Int64(value))
+            data.append(asciiDoubleQuote)
         } else {
-            append(text: String(value))
+            appendInt(value: Int64(value))
         }
     }
 
-    mutating func putBoolValue(value: Bool, quote: Bool) {
-        if quote {
-            append(text: value ? "\"true\"" : "\"false\"")
+    internal mutating func putUInt64(value: UInt64) {
+        data.append(asciiDoubleQuote)
+        appendUInt(value: value)
+        data.append(asciiDoubleQuote)
+    }
+
+    internal mutating func putUInt32(value: UInt32) {
+        if isMapKey {
+            data.append(asciiDoubleQuote)
+            appendUInt(value: UInt64(value))
+            data.append(asciiDoubleQuote)
         } else {
-            append(text: value ? "true" : "false")
+            appendUInt(value: UInt64(value))
         }
     }
-    mutating func putStringValue(value: String) {
-        let hexDigits = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "A", "B", "C", "D", "E", "F"];
-        append(text: "\"")
+
+    internal mutating func putBoolValue(value: Bool) {
+        if isMapKey {
+            data.append(asciiDoubleQuote)
+        }
+        if value {
+            data.append(contentsOf: [asciiLowerT, asciiLowerR, asciiLowerU, asciiLowerE])
```

Comment:
Some interesting numbers from `100 bool`:  The array form here does a JSON encode in about 18us.  Using a literal constant `String` bumps that up to 23us.  But a `StaticString` is about the same as the array form, so I've switched to that.  That also provides a way to use literal constants for other multi-character items as Thomas suggested elsewhere.